### PR TITLE
[Feature] [Vectorized] Some pre-refactorings or interface additions for schema change part2

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

This pr is part of https://github.com/apache/incubator-doris/pull/9375, included some refactorings or interface additions.

1. Pass a `DescriptorTable` from` fe` to `be`.
2. Split `sc.process` to `process` and `inner_process` .
3. Change singleton class `SchemaChangeHandler` to all static function class.
4. Add a function `get_sc_procedure` wrapped some common logic.
5. Some other refactoring

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
3. Has unit tests been added: (Yes/No/No Need)
5. Has document been added or modified: (Yes/No/No Need)
6. Does it need to update dependencies: (Yes/No)
7. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/schema_change.cpp [hunk 1]
```diff
@@ -1119,17 +1080,6 @@ Status SchemaChangeWithSorting::process(RowsetReaderSharedPtr rowset_reader,
     Status res = Status::OK();
     RowsetSharedPtr rowset = rowset_reader->rowset();
 
-    if (rowset->empty() || rowset->num_rows() == 0) {
-        res = new_rowset_writer->flush();
```
Comment (compiletheworld) [inline be/src/olap/schema_change.cpp:1123]: Ditto

File: be/src/olap/schema_change.cpp [hunk 2]
```diff
@@ -1065,48 +1049,25 @@ Status SchemaChangeDirectly::process(RowsetReaderSharedPtr rowset_reader,
         return Status::OLAPInternalError(OLAP_ERR_ALTER_STATUS_ERR);
     }
 
-    // rows filtered by zone map against delete handler
-    add_filtered_rows(rowset_reader->filtered_rows());
-
-    // Check row num changes
-    if (config::row_nums_check) {
```
Comment (compiletheworld) [inline be/src/olap/schema_change.cpp:1072]: Why do we skip this check? Is this redundant or useless?

Comment (BiteTheDDDDt) [reply be/src/olap/schema_change.cpp:1072]: I move those common logic to `SchemaChange::process`, and some specific logic is placed in `_inner_process`

File: be/src/olap/schema_change.cpp [hunk 3]
```diff
@@ -1010,16 +1008,6 @@ Status SchemaChangeDirectly::process(RowsetReaderSharedPtr rowset_reader,
     }
 
     Status res = Status::OK();
-    if (rowset_reader->rowset()->empty() || rowset_reader->rowset()->num_rows() == 0) {
```
[the remark this answers is not in the document]
Comment (BiteTheDDDDt) [reply be/src/olap/schema_change.cpp:1013]: moved to `SchemaChange::process`

**Comment by compiletheworld** [comment]
LGTM

File: be/src/olap/reader.h [hunk 4]
```diff
@@ -108,7 +108,7 @@ class TabletReader {
     virtual ~TabletReader();
 
     // Initialize TabletReader with tablet, data version and fetch range.
-    virtual Status init(const ReaderParams& read_params);
+    virtual Status init(const ReaderParams& read_params, bool is_alter_table = false);
```
[the remark this answers is not in the document]
Comment (BiteTheDDDDt) [reply be/src/olap/reader.h]: done

File: be/src/olap/schema_change.cpp [hunk 5]
```diff
@@ -92,19 +91,20 @@ class RowBlockMerger {
...
+        : _desc_tbl(desc_tbl) {
     _schema_mapping.resize(tablet_schema.num_columns());
     _delete_handler = delete_handler;
 }
 
 RowBlockChanger::~RowBlockChanger() {
-    SchemaMapping::iterator it = _schema_mapping.begin();
-    for (; it != _schema_mapping.end(); ++it) {
+    for (auto it = _schema_mapping.begin(); it != _schema_mapping.end(); ++it) {
         SAFE_DELETE(it->default_value);
```
[the remark this answers is not in the document]
Comment (BiteTheDDDDt) [reply be/src/olap/schema_change.cpp:108]: `_schema_mapping.default_value` seems be inited by `_parse_request(... RowBlockChanger* rb_changer ...`, so maybe we better delete it by whole vector.

File: be/src/olap/schema_change.h [hunk 6]
```diff
@@ -94,20 +85,62 @@ class SchemaChange {
...
+        }
+
+        _filtered_rows = 0;
+        _merged_rows = 0;
+
+        RETURN_IF_ERROR(_inner_process(rowset_reader, rowset_writer, new_tablet, base_tablet));
+        _add_filtered_rows(rowset_reader->filtered_rows());
+
+        // Check row num changes
+        if (config::row_nums_check) {
```
[the remark this answers is not in the document]
Comment (BiteTheDDDDt) [reply be/src/olap/schema_change.h]: done

File: be/src/vec/columns/column.h [hunk 7]
```diff
@@ -199,14 +201,22 @@ class IColumn : public COW<IColumn> {
         LOG(FATAL) << "Method insert_many_binary_data is not supported for " << get_name();
     }
 
+    virtual void insert_many_data(const char* pos, size_t length, size_t data_num) {
```
Comment (HappenLee) [inline be/src/vec/columns/column.h:204]: the function seems not need be `virtual`

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**